components/clarinet-files/src/project_manifest.rs

```diff
@@ -243,16 +249,19 @@ impl ProjectManifest {
                                 }
                                 _ => DEFAULT_EPOCH,
                             };
-                            config_contracts.insert(
-                                contract_name.to_string(),
-                                ClarityContract {
-                                    name: contract_name.clone(),
-                                    code_source,
-                                    deployer,
-                                    clarity_version,
-                                    epoch,
-                                },
-                            );
+                            let clarity_contract = ClarityContract {
+                                name: contract_name.clone(),
+                                code_source,
+                                deployer,
+                                clarity_version,
+                                epoch,
+                            };
+                            config_contracts
+                                .insert(contract_name.clone(), clarity_contract.clone());
```

Comment:
I think there is some room for memory optimization here, with these extra-clones.
Could we add a todo / or create an issue, and get back to it when you feel more confortable with lifetimes or if we feel like performance is becoming a bottleneck?
Thanks!

Reply:
Partially addressed, more improvements to come as part of #668